### ISSUE TITLE
Error importing file: <dylib_name> when extracting from dyld_shared_cache_arm64

### ISSUE THREAD

**Issue by hot3eed** [NationalSecurityAgency/ghidra#2216]
**Describe the bug**
Ghidra fails to extract a dylib from the dyld_shared_cache_arm64. It gives an IO exception. I've tried the following dylibs, the result is the same:
* `System/Library/PrivateFrameworks/IMCore.framework/IMCore`
* `System/Library/PrivateFrameworks/IMFoundation.framework/IMFoundation`
* `System/Library/PrivateFrameworks/AVConference.framework/Frameworks/LegacyHandle.framework/LegacyHandle` (This is the smallest framework in the shared cache.)

**To Reproduce**
1. Create a new Ghidra project
2. In the menubar, click on File -> Import File
3. Choose "File System"
4. Choose a dylib.
5. Right click on it, then Import, then OK
6. See error.

**Expected behavior**
Successful extraction of the framework/dylib.

**Attachments**
Stack trace:
```
Error importing file: LegacyHandle
java.io.IOException: Offset must be in range [0,1020788735], got 1170450424
	at ghidra.app.util.opinion.MachoLoader.load(MachoLoader.java:98)
	at ghidra.app.util.opinion.AbstractLibrarySupportLoader.doLoad(AbstractLibrarySupportLoader.java:346)
	at ghidra.app.util.opinion.AbstractLibrarySupportLoader.loadProgram(AbstractLibrarySupportLoader.java:83)
	at ghidra.app.util.opinion.AbstractProgramLoader.load(AbstractProgramLoader.java:112)
	at ghidra.plugin.importer.ImporterUtilities.importSingleFile(ImporterUtilities.java:401)
	at ghidra.plugin.importer.ImporterDialog.lambda$okCallback$7(ImporterDialog.java:351)
	at ghidra.util.task.TaskLauncher$1.run(TaskLauncher.java:90)
	at ghidra.util.task.Task.monitoredRun(Task.java:126)
	at ghidra.util.task.TaskRunner.lambda$startTaskThread$1(TaskRunner.java:94)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

---------------------------------------------------
Build Date: 2020-Feb-12 1149 EST
Ghidra Version: 9.1.2
Java Home: /usr/local/Cellar/openjdk@11/11.0.7+10/libexec/openjdk.jdk/Contents/Home
JVM Version: Oracle Corporation 11.0.7
OS: Mac OS X 10.15.3 x86_64
Workstation: localhost
```

**Environment:**
 - OS: macOS 10.15.3 (19D76)
 - Java Version: openjdk 11.0.7
 - Ghidra Version: 9.1.2 PUBLIC
 - dyld_shared_cache_arm64: iPhone 6s (iOS 13.6)

**Comment by ryanmkurtz** [comment]
See #682.  There are known issues with extracting dylib's from the cache.

**Comment by ryanmkurtz** [comment]
Fixed by 147e6b8f136363a5adf40ba64d6b400304954bf1
